File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    a = 0
    b = len(people) - 1
    
    people.sort()
    
    while a < b:
        if people[a] + people[b] <= limit:
            a += 1
            answer += 1
        b -= 1
        
    return len(people) - answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        int first = 0, last = people.length - 1; 
        
        Arrays.sort(people);
        
        while (first <= last){
            if (first == last){
                answer += 1;
                break;
            }
            
            if (people[first] + people[last] <= limit){
                answer += 1;
                first += 1;
                last -= 1;
            }else{
                answer += 1;
                last -= 1;
            }
        }
        
        return answer;
    }
}